spyder/config/base.py

```diff
@@ -267,18 +267,21 @@ def get_interface_language():
     """
     locale_language = locale.getdefaultlocale()[0]

```

Comment:
William, the try except needs to be around this line, not below :-)

Besides, you only need to catch `ValueError`, which was the error reported on issue #3627.